File: Stack/Q2. Evaluate Reverse Polish Notation/evalRPN.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        vector<int> stk;

        for(auto& oper : tokens){
            if(oper != "*" && oper != "/" && oper != "+" && oper != "-"){
                stk.push_back(stoi(oper));
            }
            else{
                int sec = stk.back();
                stk.pop_back();
                int first = stk.back();
                stk.pop_back();
                int value = 0;

                if(oper == "+")value = first + sec;
                else if(oper == "-")value = first - sec;
                else if(oper == "*")value = first * sec;
                else value = first / sec;
                stk.push_back(value);
            }
        }

        return stk[0];
    }
};

File: Stack/Q2. Evaluate Reverse Polish Notation/evalRPN.js
/**
 * @param {string[]} tokens
 * @return {number}
 */
var evalRPN = function (tokens) {
    let stk = []

    for (let oper of tokens) {
        if (oper !== "+" && oper !== "-" && oper !== "/" && oper !== "*") {
            stk.push(parseInt(oper))
        }
        else {
            let second = stk.pop()
            let first = stk.pop()
            let num = 0
            switch (oper) {
                case "+":
                    num = first + second
                    break;
                case "/":
                    num = first / second
                    break;
                case "-":
                    num = first - second
                    break;
                case "*":
                    num = first * second
                    break
            }
            stk.push(parseInt(num))
        }
    }
    return stk[0]
};